volunteer_base/models/volunteer_project.py

```diff
@@ -0,0 +1,105 @@
+# © 2021 humanilog (https://www.humanilog.org)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class VolunteerProject(models.Model):
+    _name = 'volunteer.project'
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _description = 'Volunteer Projects'
+    _order = 'sequence, name, id'    
+
+    active = fields.Boolean(
+        default=True,
+        help="The active field allows you to hide the project without removing it."
+    )
+    name = fields.Char(
+        string='Name',
+        index=True,
+        required=True,
+        tracking=True,
+        track_visibility='always',
+        translate=True,
+        help="Name of the volunteer project",
+    )
+    sequence = fields.Char(
+        string='Reference',
+        default="/",
+        help="Sequence to identify volunteer projects easily",
+    )
+    description = fields.Html(
+        string='Project Description',
+        translate=True,
+    )
+    timeframe = fields.Selection(
+        string='Timeframe',
+        selection = [
+            ('ongoing', 'Ongoing'),
+            ('period', 'Time Period'),
+        ],
+        default='ongoing',
+        required=True,
+        tracking=True,
+        track_visibility='always',
+        help="Select wether the project has a conrete period with a start end end date or is ongoing",
+    )
+    date_start = fields.Date(string='Start Date'
+    )
+    date_end = fields.Date(string='End Date'
+    )
+    engagement_default_duration = fields.Integer(
+        default=1,
+        string='Default Duration',
+        help="Default duration for engagements created for this project.\n"
+             "This timeframe will be used for computation of start and end dates",
+    )
+    engagement_default_duration_type = fields.Selection([
+        ('daily', 'Day(s)'),
+        ('weekly', 'Week(s)'),
+        ('monthly', 'Month(s)')],
+        default='monthly',
+        string='Duration Type',
+        help="Specify duration type for engagement default duration.",
+    )
+    engagement_ids = fields.One2many(
+        'volunteer.engagement',
+        'volunteer_project_id',
+        string='Engagements',
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('sequence', '/') == '/':
+                vals['sequence'] = self.env['ir.sequence'].next_by_code('volunteer.project')
+        return super(VolunteerProject, self).create(vals_list)
```

Comment:
With @api.multi you have to expect, that self contains multiple objects.
Loop through self and apply your code to the single object where the loop is pointing to.

I would also suggest to create the super object first and then apply your object manipulations.